dali/kernels/algebra/linear_transformation_gpu.h

```diff
@@ -0,0 +1,198 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_ALGEBRA_LINEAR_TRANSFORMATION_GPU_H_
+#define DALI_KERNELS_ALGEBRA_LINEAR_TRANSFORMATION_GPU_H_
+
+#include <vector>
+#include "dali/core/convert.h"
+#include "dali/core/geom/box.h"
+#include "dali/kernels/common/block_setup.h"
+#include "dali/kernels/imgproc/surface.h"
+#include "dali/kernels/imgproc/roi.h"
+
+namespace dali {
+namespace kernels {
+namespace linear_transformation {
+namespace detail {
+
+using TransformationMatrixElementType = float;
+
+template <class OutputType, class InputType, size_t N, size_t M, size_t spatial_ndims>
+struct SampleDescriptor {
+  const InputType *in;
+  OutputType *out;
+  ivec<spatial_ndims> in_size, in_strides, out_size, out_strides;
+  int in_channels, out_channels;
```

Comment:
If the matrix is NxM, why do you need number of channels? Doesn't it have to be exactly in_channels == M and out_channels == N?

Reply:
True, fixed